### PR TITLE
Add comm context manager, add phi broadcast op

### PR DESCRIPTION
**Pull request by hitywt**
### PR types
Function optimization

### PR changes
OPs

### Describe
1. set comm_context for dev_context in interpreter, in order to use new comm_context_manager
2. add phi broadcast kernel and api based on comm_context

### CODE REVIEW

File: paddle/fluid/framework/new_executor/interpreter/interpreter_util.cc [hunk 1]
```diff
@@ -967,6 +969,21 @@ void LogDeviceMemoryStats(const platform::Place& place) {
...
+                          platform::DeviceContext* dev_ctx) {
+  if (operator_base->HasAttr("ring_id")) {
+    int ring_id = operator_base->Attr<int>("ring_id");
+    const auto& comm_context_manager =
+        phi::distributed::CommContextManager::GetInstance();
+    if (comm_context_manager.Has(ring_id)) {
+      auto comm_context = comm_context_manager.Get(ring_id);
+      if (!dev_ctx->GetCommContext()) {
+        dev_ctx->SetCommContext(comm_context);
+      }
```
Comment (LiYuRio) [inline paddle/fluid/framework/new_executor/interpreter/interpreter_util.cc:1188]: 这里或许后面可以加一个如果找不见的报错

File: paddle/phi/kernels/gpu/broadcast_kernel.cu [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+PD_REGISTER_KERNEL(broadcast,
+                   GPU,
+                   ALL_LAYOUT,
+                   phi::BroadcastKernel,
+                   float,
+                   double,
+                   int,
+                   int64_t,
+                   phi::dtype::float16) {}
```
Comment (LiYuRio) [inline paddle/phi/kernels/gpu/broadcast_kernel.cu]: 这里后面可以加一下bf16，int8等这种类型的支持，动态图下是已经支持了的

**Review by LiYuRio — DISMISSED** [review]
LGTM

**Review by ForFishes — DISMISSED** [review]
LGTM

**Review by aoyulong — DISMISSED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM for 
`set_tests_properties(test_collective_broadcast_api
                       PROPERTIES TIMEOUT "300" LABELS "RUN_TYPE=DIST")`

**Review by LiYuRio — APPROVED** [review]
LGTM

**Pull request merged**